termux-exec.c

```diff
@@ -231,8 +236,33 @@ int execve(const char* filename, char* const* argv, char* const* envp)
 		}
 	}
 
-	int ret = real_execve(filename, argv, envp);
+	int ret = real_exec_fn(filename, argv, envp);
 	free(new_argv);
 	free(new_envp);
 	return ret;
+	
+}
+
+int _execvp(const char* filename, char* const* argv, char* const* envp)
+{
+        (void)envp;
+        int (*real_execvp) (const char*, char* const[]) = dlsym(RTLD_NEXT, "execvp");
+        return real_execvp(filename, argv);;
```

Comment:
We don't do any special treatment here. Then why bother intercepting this weak symbol in the first place?

Reply:
Because in the exec_wrapper, the real_exec_fn() function, which takes three parameters, will be passed in. The actual execvp function has one fewer parameter than execve, as it lacks the envp parameter. Therefore, I wrapped the original functions in separate functions so that they all have three parameters in form. Perhaps the names I chose aren't ideal, as the intention here is not to intercept the original system functions.